contracts/swap/src/swap.rs

```diff
@@ -131,57 +135,73 @@ pub fn handle_atomic_order_reply(
+
+    // need to remove protobuf scale factor to get real values
+    let average_price = FPDecimal::from_str(&trade_data.price)? / dec_scale_factor;
+    let quantity = FPDecimal::from_str(&trade_data.quantity)? / dec_scale_factor;
     let fee = FPDecimal::from_str(&trade_data.fee)? / dec_scale_factor;
-    deps.api.debug(&format!(
-        "Quantity: {quantity}, price {avg_price}, fee {fee}"
-    ));
+
+    let mut swap_results = SWAP_RESULTS.load(deps.storage)?;
 
     let current_step = STEP_STATE.load(deps.storage).map_err(ContractError::Std)?;
+
     let new_quantity = if current_step.is_buy {
         quantity
     } else {
-        quantity * avg_price - fee
+        quantity * average_price - fee
     };
 
     let new_balance = FPCoin {
         amount: new_quantity,
         denom: current_step.step_target_denom,
     };
 
-    deps.api.debug(&format!("New balance: {new_balance:?}"));
-
     let swap = SWAP_OPERATION_STATE.load(deps.storage)?;
+
+    swap_results.push(SwapResults {
+        market_id: swap.swap_steps[(current_step.step_idx) as usize].to_owned(),
+        price: average_price,
+        quantity,
+        fee,
+    });
+
     if current_step.step_idx < (swap.swap_steps.len() - 1) as u16 {
-        execute_swap_step(deps, env, swap, current_step.step_idx + 1, new_balance)
-            .map_err(ContractError::Std)
-    } else {
-        // last step, finalize and send back funds to a caller
-        if new_balance.amount < swap.min_target_quantity {
-            return Err(ContractError::MinExpectedSwapAmountNotReached(
-                swap.min_target_quantity,
-            ));
-        }
-        let send_message = BankMsg::Send {
-            to_address: swap.sender_address.to_string(),
-            amount: vec![new_balance.clone().into()],
-        };
-        deps.api.debug(&format!("Send message: {send_message:?}"));
-        SWAP_OPERATION_STATE.remove(deps.storage);
-        STEP_STATE.remove(deps.storage);
-        let response = Response::new()
-            .add_message(send_message)
-            .add_attribute("swap_final_amount", new_balance.amount.to_string())
-            .add_attribute("swap_final_denom", new_balance.denom);
-
-        Ok(response)
+        SWAP_RESULTS.save(deps.storage, &swap_results)?;
+        return execute_swap_step(deps, env, swap, current_step.step_idx + 1, new_balance)
+            .map_err(ContractError::Std);
     }
+
+    // last step, finalize and send back funds to a caller
+    if new_balance.amount < swap.min_target_quantity {
+        return Err(ContractError::MinExpectedSwapAmountNotReached(
+            swap.min_target_quantity,
+        ));
+    }
+    let send_message = BankMsg::Send {
+        to_address: swap.sender_address.to_string(),
+        amount: vec![new_balance.clone().into()],
+    };
+
+    let swap_results_json = serde_json_wasm::to_string(&swap_results).unwrap();
+    let swap_event = Event::new("atomic_swap_execution")
```

Comment:
is that the most correct event name? maybe `injective_converter_execution` would be better?

Reply:
`atomic_swap_execution` is good I'd say, it shouldn't have injective_ as prefix since we might have other contracts build on us and can use same event